processFlows.json

```diff
@@ -1546,5 +1546,532 @@
+          }
+        }
+      },
+      { "op": "And" },
+      {
+        "restriction": {
+          "MatchInputOutputMetadataValue": {
+            "inputIndex": 1,
+            "inputMetadataKey": "originalId",
+            "outputIndex": 2,
+            "outputMetadataKey": "demandB"
+          }
+        }
+      },
+      { "op": "And" },
+      {
+        "restriction": {
+          "MatchInputOutputMetadataValue": {
+            "inputIndex": 2,
+            "inputMetadataKey": "demandB",
+            "outputIndex": 2,
+            "outputMetadataKey": "demandB"
+          }
+        }
+      },
+      { "op": "And" },
+      {
+        "restriction": {
+          "MatchInputOutputMetadataValue": {
+            "inputIndex": 2,
+            "inputMetadataKey": "originalId",
+            "outputIndex": 2,
+            "outputMetadataKey": "originalId"
+          }
+        }
+      },
+      {
+        "restriction": {
+          "InputHasMetadata": {
+            "index": 2,
+            "metadataKey": "originalId"
+          }
+        }
+      },
+      {
+        "restriction": {
+          "None": {}
+        }
+      },
+      { "op": "NotL" },
+      {
+        "restriction": {
+          "MatchInputIdOutputMetadataValue": {
+            "inputIndex": 2,
+            "outputIndex": 2,
+            "outputMetadataKey": "originalId"
+          }
+        }
+      },
+      { "op": "And" },
+      { "op": "Or" },
+      { "op": "And" },
+      {
+        "restriction": {
+          "FixedInputMetadataValue": {
+            "index": 2,
+            "metadataKey": "state",
+            "metadataValue": {
+              "Literal": "acceptedFinal"
+            }
+          }
+        }
+      },
+      { "op": "And" },
+      {
+        "restriction": {
+          "FixedOutputMetadataValue": {
+            "index": 2,
+            "metadataKey": "state",
+            "metadataValue": {
+              "Literal": "cancelled"
+            }
+          }
+        }
+      },
+      { "op": "And" },
+      {
+        "restriction": {
+          "FixedInputMetadataValue": {
+            "index": 2,
+            "metadataKey": "state",
+            "metadataValue": {
+              "Literal": "acceptedFinal"
+            }
+          }
+        }
+      },
+      {
+        "restriction": {
+          "SenderHasOutputRole": {
+            "index": 2,
+            "roleKey": "MemberA"
+          }
+        }
+      },
+      { "op": "Or" },
+      {
+        "restriction": {
+          "FixedInputMetadataValue": {
+            "index": 2,
+            "metadataKey": "state",
+            "metadataValue": {
+              "Literal": "acceptedFinal"
+            }
+          }
+        }
+      },
+      {
+        "restriction": {
+          "SenderHasOutputRole": {
+            "index": 2,
+            "roleKey": "MemberB"
+          }
+        }
+      },
+      { "op": "And" },
+      { "op": "Or" },
+      { "op": "And" }
```

Comment:
how do i specify what "or" relates to/encompasses ?

Reply:
It's a stack based evaluation. Each time we mention a `restriction` we push the value of that restriction onto the stack. Each `op` pulls two values of the stack. So lets say we have something like

```js
[
  RestrictionA,
  RestrictionB,
  RestrictionC,
  Or,
  And
]
```

The evaluation will then go like:

1. `stack=[RestrictionA]`
2. `stack=[RestrictionA, RestrictionB]`
3. `stack=[RestrictionA, RestrictionB, RestrictionC]`
4. `stack=[RestricitonA, (RestrictionB || RestrictionC)]`
5. `stack=[RestricitonA && (RestrictionB || RestrictionC)]` 

So at the end the result is `RestricitonA && (RestrictionB || RestrictionC)`

Reply:
So you're close but not quite there. Have another look and if you're still stuck we can talk it through

Reply:
So I think it will be like this: 
```
    {
        "restriction": {
          "FixedInputMetadataValue": {
            "index": 2,
            "metadataKey": "state",
            "metadataValue": {
              "Literal": "acceptedFinal"
            }
          }
        }
      },
      { "op": "And" },
      {
        "restriction": {
          "FixedOutputMetadataValue": {
            "index": 2,
            "metadataKey": "state",
            "metadataValue": {
              "Literal": "cancelled"
            }
          }
        }
      },
      { "op": "And" },
      {
        "restriction": {
          "SenderHasOutputRole": {
            "index": 2,
            "roleKey": "MemberA"
          }
        }
      },
      {
        "restriction": {
          "SenderHasOutputRole": {
            "index": 2,
            "roleKey": "MemberB"
          }
        }
      },
      { "op": "Or" },
      { "op": "And" } 
```

Reply:
Yes that looks correct to me